src/Sentry/Internal/Http/RetryAfterHandler.cs

```diff
@@ -63,31 +64,43 @@ internal RetryAfterHandler(HttpMessageHandler innerHandler, ISystemClock clock)
 
             var response = await base.SendAsync(request, cancellationToken).ConfigureAwait(false);
 
-            if (response.StatusCode == TooManyRequests && response.Headers != null)
+            if (response.StatusCode == TooManyRequests)
+            {
+                var retryAfterTimestamp = GetRetryAfterTimestamp(response);
+                _ = Interlocked.Exchange(ref _retryAfterUtcTicks, retryAfterTimestamp.UtcTicks);
+            }
+
+            return response;
+        }
+
+        private DateTimeOffset GetRetryAfterTimestamp(HttpResponseMessage response)
+        {
+            if (response.Headers != null)
             {
                 if (response.Headers.RetryAfter != null)
                 {
-                    if (response.Headers.RetryAfter.Delta != null)
+                    if (response.Headers.RetryAfter.Delta is { } delta)
                     {
-                        var retryAfterUtc = _clock.GetUtcNow() + response.Headers.RetryAfter.Delta.Value;
-                        _ = Interlocked.Exchange(ref _retryAfterUtcTicks, retryAfterUtc.UtcTicks);
+                        return _clock.GetUtcNow() + delta;
                     }
-                    else if (response.Headers.RetryAfter.Date != null)
+
+                    if (response.Headers.RetryAfter.Date is { } date)
                     {
-                        _ = Interlocked.Exchange(ref _retryAfterUtcTicks, response.Headers.RetryAfter.Date.Value.UtcTicks);
+                        return date;
                     }
                 }
+
                 // Sentry was sending floating point numbers which are not handled by RetryConditionHeaderValue
                 // To be compatible with older versions of sentry on premise: https://github.com/getsentry/sentry/issues/7919
-                else if (response.Headers.TryGetValues("Retry-After", out var values)
-                         && double.TryParse(values?.FirstOrDefault(), out var retryAfterSeconds))
+                if (response.Headers.TryGetValues("Retry-After", out var values)
```

Comment:
TryGetValues has a notnullwhentrue for value. so the following `values?` is redundant?